Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs

```diff
@@ -1,148 +1,179 @@
+        [Dependency] private readonly IRobustRandom _random;
+        [Dependency] private readonly IEntitySystemManager _entitySystem;
 #pragma warning restore 649
+
+        private AudioSystem _audioSystem;
         public override string Name => "Drink";
+
         [ViewVariables]
         private SolutionComponent _contents;
-
-        private AppearanceComponent _appearanceComponent;
-
         [ViewVariables]
         private string _useSound;
         [ViewVariables]
-        private string _finishPrototype;
-
-        public ReagentUnit TransferAmount => _transferAmount;
+        private bool _defaultToOpened;
         [ViewVariables]
-        private ReagentUnit _transferAmount = ReagentUnit.New(2);
-
-        public ReagentUnit MaxVolume
-        {
-            get => _contents.MaxVolume;
-            set => _contents.MaxVolume = value;
-        }
-
-        private bool _despawnOnFinish;
+        public ReagentUnit TransferAmount { get; private set; } = ReagentUnit.New(2);
 
-        private bool _drinking;
+        [ViewVariables]
+        public bool Opened => _opened;
 
-        public int UsesLeft()
-        {
-            // In case transfer amount exceeds volume left
-            if (_contents.CurrentVolume == 0)
-            {
-                return 0;
-            }
-            return Math.Max(1, (int)Math.Ceiling((_contents.CurrentVolume / _transferAmount).Float()));
-        }
+        [ViewVariables]
+        public bool Empty => _contents.CurrentVolume.Float() <= 0;
 
+        private AppearanceComponent _appearanceComponent;
+        private bool _opened = false;
+        private string _soundCollection;
+        private string _originalName;
 
         public override void ExposeData(ObjectSerializer serializer)
         {
             base.ExposeData(serializer);
-            serializer.DataField(ref _useSound, "use_sound", "/Audio/items/drink.ogg");
-            // E.g. cola can when done or clear bottle, whatever
-            // Currently this will enforce it has the same volume but this may change. - TODO: this should be implemented in a separate component
-            serializer.DataField(ref _despawnOnFinish, "despawn_empty", false);
-            serializer.DataField(ref _finishPrototype, "spawn_on_finish", null);
+            serializer.DataField(ref _useSound, "useSound", "/Audio/items/drink.ogg");
+            serializer.DataField(ref _defaultToOpened, "isOpen", false); //For things like cups of coffee.
+            serializer.DataField(ref _soundCollection, "openSounds","canOpenSounds");
         }
 
-        protected override void Startup()
+        public override void Initialize()
         {
-            base.Startup();
-            _contents = Owner.GetComponent<SolutionComponent>();
+            base.Initialize();
+            Owner.TryGetComponent(out _appearanceComponent);
+            Owner.TryGetComponent<SolutionComponent>(out _contents);
```

Comment:
Why did you change to using `TryGet` instead of just `Get`?

Reply:
oh, i did. wtf.

Reply:
it looks better and doesnt crash the game if a drink bottle is doesnt have a solutioncomponent

Reply:
But `_contents` is used in a non-null way.

Reply:
This should be fixed, yes.